src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/NocRespondentRepresentativeService.java

```diff
@@ -162,4 +173,34 @@ public List<ChangeOrganisationRequest> getRepresentationChanges(CaseData  after,
 
         return changeRequests;
     }
+
+    /**
+     * Revokes access from all users of an organisation being replaced or removed.
+     * @param caseId - case id of case to apply update to
+     * @param changeOrganisationRequest - containing case role and id of organisation to remove
+     * @throws IOException - thrown if no ccd service is inaccessible
+     */
+    public void removeOrganisationRepresentativeAccess(String caseId,
+                                                       ChangeOrganisationRequest changeOrganisationRequest)
+        throws IOException {
+        String roleOfRemovedOrg = changeOrganisationRequest.getCaseRoleId().getSelectedCode();
+        String orgId = changeOrganisationRequest.getOrganisationToRemove().getOrganisationID();
+        CaseUserAssignmentData caseAssignments =
+            nocCcdService.getCaseAssignments(adminUserService.getAdminUserToken(), caseId);
+
+        List<CaseUserAssignment> usersToRevoke = caseAssignments.getCaseUserAssignments().stream()
+            .filter(caseUserAssignment -> caseUserAssignment.getCaseRole().equals(roleOfRemovedOrg))
+            .map(caseUserAssignment ->
+                CaseUserAssignment.builder().userId(caseUserAssignment.getUserId())
+                    .organisationId(orgId)
+                    .caseRole(roleOfRemovedOrg)
+                    .caseId(caseId)
+                    .build()
+            ).collect(toList());
+
+        if (!CollectionUtils.isEmpty(usersToRevoke)) {
```

Comment:
Not gonna block but I believe we have a method for isNotEmpty rather than inversing isEmpty